src/ACEtk/block/IncoherentScatteringFunctionBlock.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef NJOY_ACETK_BLOCK_INCOHERENTSCATTERINGFUNCTIONBLOCK
+#define NJOY_ACETK_BLOCK_INCOHERENTSCATTERINGFUNCTIONBLOCK
+
+// system includes
+#include <algorithm>
+
+// other includes
+#include "ACEtk/block/details/ArrayData.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+
+/**
+ *  @class
+ *  @brief The photoatomic JINC block with the incoherent scattering function values
+ *
+ *  The IncoherentScatteringFunctionBlock class contains the scattering functions
+ *  tabulated on a fixed grid of the recoil electron momentum transfer (in inverse
+ *  angstroms).
+ */
+class IncoherentScatteringFunctionBlock : protected details::ArrayData {
+
+  /* fields */
+
+  /* auxiliary functions */
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/IncoherentScatteringFunctionBlock/src/ctor.hpp"
+
+  /* methods */
+
+  /**
+   *  @brief Return the number of values
+   */
+  constexpr unsigned int NM() const { return 21; }
```

Comment:
Same here, if the difference in JXS is != 21, it assumes there's two arrays with the first containing momentum.